csharp/src/SeedLang.X/PythonVisitor.cs

```diff
@@ -0,0 +1,102 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Diagnostics;
+using Antlr4.Runtime;
+using Antlr4.Runtime.Misc;
+using SeedLang.Ast;
+
+namespace SeedLang.X {
+  // The visitor class to visit each SeedPython parse tree and generate the corresponding AST tree.
+  //
+  // The default implement of SeedPythonBaseVisitor is to visit all the children and return the
+  // result of the last one. PythonVisitor overrides the method if the default implement is not
+  // correct.
+  internal class PythonVisitor : SeedPythonBaseVisitor<AstNode> {
+    // Visits add and subtract binary expressions.
+    //
+    // The expr() method of the Add_subContext return a ExprContext array which contains exact 2
+    // items: the left and right ExprContexts.
+    public override AstNode VisitAdd_sub([NotNull] SeedPythonParser.Add_subContext context) {
+      return BuildBinary(context.op, context.expr());
+    }
+
+    // Visits multiply and divide binary expressions.
+    //
+    // The expr() method of the Add_subContext return a ExprContext array which contains exact 2
+    // items: the left and right ExprContexts.
+    public override AstNode VisitMul_div([NotNull] SeedPythonParser.Mul_divContext context) {
+      return BuildBinary(context.op, context.expr());
+    }
+
+    // Visits number expression.
+    public override AstNode VisitNumber([NotNull] SeedPythonParser.NumberContext context) {
+      return Expression.Number(double.Parse(context.GetText()));
+    }
+
+    // Visits grouping expression.
+    //
+    // There isn't corresponding grouping AST node. The order of the expression node in the AST tree
+    // represent the grouping structure.
+    public override AstNode VisitGrouping([NotNull] SeedPythonParser.GroupingContext context) {
+      return Visit(context.expr());
+    }
+
+    // Visits the simple statement.
+    //
+    // The small_stmt() method of the Simple_stmtContext returns a array which contains all the
+    // small statements. There is at least one small statement in it.
+    //
+    // TODO: parse all the small statements in it, only parse the first one now.
+    public override AstNode VisitSimple_stmt(
+      [NotNull] SeedPythonParser.Simple_stmtContext context) {
+      SeedPythonParser.Small_stmtContext[] smallStatements = context.small_stmt();
+      Debug.Assert(smallStatements.Length > 0);
+      return Visit(smallStatements[0]);
+    }
+
+    // Visits the eval statement.
```

Comment:
the eval statement. -> eval statements.

Please make the article usage consistent across all the method comments in this class. I think either "Visits a ... expression" or "Visits ... expressions" is okay here.

Reply:
Done